### ISSUE TITLE
Implement Multi-Turn Chat with Discourse Model and AST Context

### ISSUE THREAD

**Issue by marctjones** [marctjones/klareco#694]
## Goal
Enable Klareco to maintain coherent multi-turn conversations by tracking discourse context as an **AST forest** instead of a text context window.

## Motivation
Traditional LLMs use raw text context windows. Klareco should track conversation as structured ASTs, enabling:
- Deterministic coreference resolution
- Discourse relation understanding
- Entity tracking across turns
- Explainable conversation state

### ChatManager
```python
class ChatManager:
    """Maintain conversation context as AST forest"""
    
    def __init__(self):
        self.conversation_history = []  # List of ASTs
        self.coreference_tracker = CoreferenceTracker()
        self.discourse_model = DiscourseModel()
    
    def chat(self, user_input: str) -> str:
        # Parse user input
        user_ast = self.parser.parse(user_input)
        
        # Resolve coreferences using previous ASTs
        resolved_ast = self.coreference_tracker.resolve(
            user_ast, 
            context=self.conversation_history
        )
        
        # Understand discourse relation
        relation = self.discourse_model.infer_relation(
            current=resolved_ast,
            previous=self.conversation_history[-1] if self.conversation_history else None
        )
        
        # Generate response based on discourse context
        response_ast = self.generate_response(resolved_ast, relation)
        
        # Update history
        self.conversation_history.extend([user_ast, response_ast])
        
        # Deparse
        return deparse(response_ast)
```

### CoreferenceTracker (Deterministic + Learned)
```python
class CoreferenceTracker:
    """Track entities across conversation using AST structure"""
    
    def resolve(self, current_ast, context):
        # Deterministic rules for Esperanto pronouns
        if current_ast['subjekto']['radiko'] in ['li', 'ŝi', 'ĝi', 'ili']:
            pronoun = current_ast['subjekto']['radiko']
            
            # Look back in context for matching entity
            for prev_ast in reversed(context):
                if self._matches_gender_number(prev_ast['subjekto'], pronoun):
                    # Replace pronoun with referent
                    current_ast['subjekto'] = prev_ast['subjekto']
                    break
        
        return current_ast
```

### DiscourseModel (Minimal Learned Component, ~10M params)
```python
class DiscourseModel(nn.Module):
    """Predict discourse relation between conversation turns"""
    
    RELATIONS = [
        'continuation',   # User continues previous topic
        'elaboration',    # User asks for more details
        'question',       # User asks new question
        'answer',         # User answers AI's question
        'correction',     # User corrects AI's statement
        'topic_shift',    # User changes topic
    ]
    
    def forward(self, current_ast, previous_ast):
        current_emb = self.tree_encoder(current_ast)
        previous_emb = self.tree_encoder(previous_ast)
        
        # Compute relation
        relation_logits = self.classifier(
            torch.cat([current_emb, previous_emb, current_emb * previous_emb], dim=-1)
        )
        
        return self.RELATIONS[relation_logits.argmax()]
```

### Phase 1: Deterministic Coreference Resolution (0 params)
- [ ] Implement pronoun resolution rules for Esperanto
- [ ] Track entities across conversation turns
- [ ] Test on simple dialogues

### Phase 2: Discourse Model (~10M params)
- [ ] Collect multi-turn conversation data in Esperanto
- [ ] Annotate discourse relations (6 classes)
- [ ] Train TreeLSTM-based discourse classifier
- [ ] Evaluate on held-out dialogues

### Phase 3: ChatManager Integration
- [ ] Implement conversation state management
- [ ] Integrate coreference tracker + discourse model
- [ ] Add response generation based on discourse context
- [ ] Test multi-turn coherence

### Phase 4: CLI and Evaluation
- [ ] Add CLI: `klareco chat --interactive`
- [ ] Evaluation metrics: coherence, entity tracking accuracy
- [ ] User study: compare to baseline (no discourse model)

### Multi-Turn Dialogues
```jsonl
{"conversation": [
  {"speaker": "user", "text": "Kiu fondis Esperanton?", "ast": {...}},
  {"speaker": "ai", "text": "Zamenhof fondis Esperanton.", "ast": {...}},
  {"speaker": "user", "text": "Kiam li naskiĝis?", "ast": {...}, "relation": "elaboration", "coreference": {"li": "Zamenhof"}},
  {"speaker": "ai", "text": "Li naskiĝis en 1859.", "ast": {...}},
  {"speaker": "user", "text": "Kie?", "ast": {...}, "relation": "continuation", "implicit_subject": "Zamenhof", "implicit_verb": "naskiĝis"}
]}
```

### Data Collection Strategy
1. Translate existing dialogue datasets to Esperanto
2. Synthetic generation using templates
3. Human annotation of discourse relations
4. Quality check: parse rate >90%

## Success Criteria
- [ ] Coreference resolution accuracy >90% on Esperanto pronouns
- [ ] Discourse relation classification accuracy >80%
- [ ] Multi-turn coherence (human evaluation) >85%
- [ ] Entity tracking: No lost referents across 5+ turns
- [ ] Training script: `scripts/train_discourse_model.sh`
- [ ] CLI: `klareco chat --interactive`

## Dependencies
- Parser (already exists)
- TreeLSTMEncoder (already exists)
- InstructionHandler (#693)
- ASTGenerator (#692)
- **Extends** #688: Coreference resolution (M2.2)

## Related Issues
- #688: M2.2 Coreference Resolution (deterministic component)
- #693: Instruction following (single-turn)
- #692: AST generation (response generation)
- #661: RST discourse relations (different focus, but related)

## Training Pipeline Position
```
Conversation History (ASTs) → [Discourse Model] → Response Generation → New AST → Update History
                            ↓
                    Coreference Tracker
```

## Conversation Example
```
User: Kiu fondis Esperanton?
AI: Zamenhof fondis Esperanton.

User: Kiam li naskiĝis?
      ↓ Coreference: li → Zamenhof
      ↓ Discourse: elaboration (ask more about Zamenhof)
AI: Li naskiĝis en 1859.

User: Kie?
      ↓ Implicit: Kie [Zamenhof naskiĝis]?
      ↓ Discourse: continuation
AI: En Białystok, Pollando.
```

## Explainability
Chat state includes:
- Full AST forest (all previous turns)
- Coreference chains (who refers to what)
- Discourse relation history
- Entity salience scores

User can inspect: `klareco chat --show-state`

## Future Extensions
- Multi-party conversations
- Long-term memory (summarize old turns)
- Proactive clarification questions
- Emotion/sentiment tracking via AST features

**Comment by marctjones** [comment]
Closing as part of a 2026-09-08 backlog cleanup (see PLAN.md, DESIGN.md, CLAUDE.md's merge gate).

This issue predates the merge gate (adopted 2026-07-13) and the AST-native orchestrator rewrite. It describes work against the pre-orchestrator RAG pipeline or the learned model stack (M1/M2 embeddings, compositional embeddings, entity classifiers, neural rerankers) that was pruned from the repo -- klareco/embeddings/, klareco/models/, and klareco/summarization/ no longer exist. It has no milestone and was never benchmarked, matching exactly the failure mode CLAUDE.md documents (25 capability issues filed 2026-05-26, 8 implemented, none benchmarked).

Labeled archived-2026-09 so this batch is recoverable: gh issue list --state closed --label archived-2026-09. If this capability is still wanted, re-file it against the current architecture naming the benchmark metric and test the merge gate requires.
